app/src/main/java/com/example/onsalestore/activities/PostDetailActivity.java

```diff
@@ -0,0 +1,119 @@
+        super.onCreate(savedInstanceState);
+        setTitle("");
+        setContentView(R.layout.activity_post_detail);
+
+        Intent intent = getIntent();
+        String action = intent.getAction();
+        Uri data = intent.getData();
+
+        postDetailImage = findViewById(R.id.postDetailImage);
+        postDetailUserName = findViewById(R.id.postDetailUserName);
+        postDetailNumberOfLikes = findViewById(R.id.postDetailNumberOfLikes);
+        postDetailNumberOfComments = findViewById(R.id.postDetailNumberOfComments);
+        rvPostComments = findViewById(R.id.rvPostComments);
+
+        allComments = new ArrayList<>();
+        commentAdapter = new CommentAdapter(this, allComments);
+        rvPostComments.setAdapter(commentAdapter);
+        rvPostComments.setLayoutManager(new LinearLayoutManager(this));
+        rvPostComments.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));
+
+        PostItem postItem = Parcels.unwrap(getIntent().getParcelableExtra("EXTRA_ITEM"));
+        String postImage = postItem.getItemImageUrl();
+
+        if (postImage != null) {
+            Glide.with(this).load(postImage).into(postDetailImage);
+        }
+
+        postDetailUserName.setText(postItem.getUser().getUsername());
+        JSONArray jsonArray = postItem.getJSONArray("comments");
+        Integer likes = postItem.getNumberOfLikes();
+        postDetailNumberOfLikes.setText(Integer.toString(likes));
+        if (jsonArray == null) {
+            postDetailNumberOfComments.setText("0");
+        } else {
+            postDetailNumberOfComments.setText(Integer.toString(jsonArray.length()));
+        }
+
+        queryComments();
+    }
+
+    private void queryComments() {
+        PostItem postItem = Parcels.unwrap(getIntent().getParcelableExtra("EXTRA_ITEM"));
+        JSONArray jsonArray = postItem.getJSONArray("comments");
+        if (jsonArray == null) {
+            return;
+        }
+        for (int i = 0; i < jsonArray.length(); ++i) {
+            try {
+                JSONObject comment = (JSONObject) jsonArray.get(i);
+                String commentID = (String) comment.get("objectId");
+                queryComment(commentID);
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private void queryComment(String commentId) {
+        ParseQuery<CommentItem> query = ParseQuery.getQuery(CommentItem.class);
+        query.addAscendingOrder("createdAt");
+        query.getInBackground(commentId, new GetCallback<CommentItem>() {
+            @Override
+            public void done(CommentItem object, ParseException e) {
+                if (e == null) {
+                    allComments.add(object);
+                    String comment = (String) object.getString("comment");
+                    Log.d("PostDetailActivity", "Comment : " + comment);
+                    commentAdapter.notifyItemInserted(allComments.size());
```

Comment:
A RecyclerView is 0-indexed, meaning the first item is at position 0, so here you should be subtracting 1.
```suggestion
                    commentAdapter.notifyItemInserted(allComments.size() - 1);
```